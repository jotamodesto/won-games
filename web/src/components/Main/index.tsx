import * as S from './styles'

export type MainProps = {
  title: string
}

const Main = ({ title }: MainProps) => {
  return (
    <S.Main>
      <h1>{title}</h1>
    </S.Main>
  )
}

export default Main
