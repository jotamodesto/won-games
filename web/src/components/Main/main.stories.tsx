import type { Meta, Story } from '@storybook/react/types-6-0'

import Main, { MainProps } from '.'

const meta = {
  title: 'Main',
  component: Main,
  args: {
    title: 'Boilerplate setup'
  },
  argTypes: {
    title: { control: { type: 'text' } }
  }
} as Meta
export default meta

export const Basic: Story<MainProps> = (args) => <Main {...args} />
