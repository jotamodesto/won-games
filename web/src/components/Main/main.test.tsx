import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main title="Boilerplate setup" />)

    expect(
      screen.getByRole('heading', { name: /boilerplate setup/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
