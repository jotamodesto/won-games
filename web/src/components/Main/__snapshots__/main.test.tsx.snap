// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.emotion-0 {
  padding: 1em;
}

<main
  class="emotion-0 emotion-1"
>
  <h1>
    Boilerplate setup
  </h1>
</main>
`;
